File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/SeriesItem.java

package com.noblemajesty.marvel.models.getCharacters;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class SeriesItem implements Parcelable
{

  @SerializedName("resourceURI")
  @Expose
  private String resourceURI;
  @SerializedName("name")
  @Expose
  private String name;
  public final static Parcelable.Creator<SeriesItem> CREATOR = new Creator<SeriesItem>() {


    @SuppressWarnings({
        "unchecked"
    })
    public SeriesItem createFromParcel(Parcel in) {
      return new SeriesItem(in);
    }

    public SeriesItem[] newArray(int size) {
      return (new SeriesItem[size]);
    }

  };

  protected SeriesItem(Parcel in) {
    this.resourceURI = ((String) in.readValue((String.class.getClassLoader())));
    this.name = ((String) in.readValue((String.class.getClassLoader())));
  }

  public SeriesItem() {
  }

  public String getResourceURI() {
    return resourceURI;
  }

  public String getName() {
    return name;
  }

  public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(resourceURI);
    dest.writeValue(name);
  }

  public int describeContents() {
    return 0;
  }

}

File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/SeriesItem.kt
package com.noblemajesty.marvel.models.getCharacters

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class SeriesItem : Parcelable {

    @SerializedName("resourceURI")
    @Expose
    val resourceURI: String
    @SerializedName("name")
    @Expose
    val name: String

    protected constructor(`in`: Parcel) {
        this.resourceURI = `in`.readValue(String::class.java.classLoader) as String
        this.name = `in`.readValue(String::class.java.classLoader) as String
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(resourceURI)
        dest.writeValue(name)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<SeriesItem> = object : Parcelable.Creator<SeriesItem> {


            override fun createFromParcel(`in`: Parcel): SeriesItem {
                return SeriesItem(`in`)
            }

            override fun newArray(size: Int): Array<SeriesItem?> {
                return arrayOfNulls(size)
            }

        }
    }

}
